pkg/sql/colexec/parallel_unordered_synchronizer.go

```diff
@@ -220,8 +221,13 @@ func (s *ParallelUnorderedSynchronizer) init(ctx context.Context) {
 				input.ToClose.CloseAndLogOnErr(ctx, "parallel unordered synchronizer input")
 			}()
 			sendErr := func(err error) {
-				if strings.Contains(err.Error(), context.Canceled.Error()) && atomic.LoadInt32(&internalCancellation) == 1 {
-					// Don't propagate an internal cancellation error.
+				ctxCanceled := strings.Contains(err.Error(), context.Canceled.Error())
```

Comment:
NB: These checks are still unsafe. Can't you use `errors.Is` here?

If not please file an issue to properly serialize errors and add a TODO here.